pkg/util/pullsecret/pullsecret.go

```diff
@@ -37,7 +37,12 @@ func UnmarshalSecretData(ps *corev1.Secret) (map[string]string, error) {
 
 	secretData := map[string]string{}
 	for k, v := range pullSecretData.Auths {
-		secretData[k] = v["auth"].(string)
+		if v["auth"] != nil {
+			secretData[k] = v["auth"].(string)
+		} else {
+			secretData[k] = ""
```

Comment:
What's the point of keeping the empty record vs just dropping it?  It's not usable.

Reply:
That's true, we clean them up anyway if they are no good. I'll remove the else statement.